### PR TITLE
Add test ID field with input validation

### LINKED ISSUE #5: Add test ID field with input validation
Add a test ID field with a few of of the following validation checks:
- Ask user to confirm before overwriting an existing test ID
- Provide user with an unused test ID on page load
- Allow user to load metadata from existing test ID

### PR DESCRIPTION
**Pull request by lealal**
1. You can now load entries from view data and from data log using the test id.
2. You can now update loaded entries. A confirmation message appears before overwriting the data

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
 venv/
 *.pyc
-trackside_ops/settings.py
+/trackside_ops/settings.py
```
Comment (hocally) [inline .gitignore:3]: @lealal This looks weird, does it work?

File: data_entry/templates/data_entry/createuser.html [hunk 2]
```diff
@@ -40,11 +41,11 @@ <h1>Create User</h1>
                 <p>Subteam: </p>
                 <input type="text" name="subteam" placeholder="Subteam" required>
                 <p>Phone: </p>
-                <input type="text" name="phone" placeholder="Subteam" required>
+                <input type="text" name="phone" placeholder="Phone" pattern="([0-9]{3}-[0-9]{3}-[0-9]{4})|[0-9]{10}" required>
```
Comment (hocally) [inline data_entry/templates/data_entry/createuser.html:44]: Niiiiice

File: data_entry/views.py [hunk 3]
```diff
@@ -1,17 +1,103 @@
...
+                    'date': date,
+                    }
+                return render(request, 'data_entry/collectdata.html', context)
+            # entry is being updated
+            else:
+                entry_id = request.POST['entry_id']
+                entry = Test.objects.get(id=entry_id)
+
+                # update fields
+                entry.weather = request.POST['weather']
```
Comment (hocally) [inline data_entry/views.py:55]: @lealal I love this, makes adding new parameters as easy as copying one of these lines and adding the new field

File: data_entry/views.py [hunk 4]
```diff
@@ -70,24 +156,57 @@ def collectData(request):
...
         method = request.POST['method']
+        # delete entry from database
         if method == "delete":
             entry_id = request.POST['entry_id']
             Test.objects.filter(id=entry_id).delete()
             entries = Test.objects.all()
             context = {'entries': entries}
             return render(request, 'data_entry/viewdata.html', context)
         
+        # filter entries
```
Comment (hocally) [inline data_entry/views.py:179]: @lealal Is this supposed to be commented out?

**Review by hocally — APPROVED** [review]
Looks awesome, excited to see this work on the prod server

File: .gitignore (hunk 1 above)
Comment (lealal) [reply .gitignore:3]: No, and the other one didn't work either. Apparently you can add settings.py to .gitignore lol I've seen people doing settings.py and local_settings.py

File: data_entry/views.py (hunk 4 above)
Comment (lealal) [reply data_entry/views.py:179]: Yes, I doubt that code will work. It is the outline of how to filter entries based on a search parameter. I think Johnny was going to work on that, so he can use this as a guideline

File: data_entry/views.py (hunk 3 above)
Comment (lealal) [reply data_entry/views.py:55]: Right!?

**Pull request merged**